content/docs/tasks/telemetry/export-logs-through-stackdriver.md

```diff
@@ -0,0 +1,301 @@
+in Istio.
+
+### Setting up various log sinks
+
+Common setup for all sinks:
+
+1. Enable StackDriver Monitoring API for the project
+1. Make sure `principalEmail` that would be setting up the sink has write access
+to the project and Logging Admin role permissions.
+
+#### BigQuery
+
+1.  Create a BigQuery dataset where you would like logs to get exported in
+    BigQuery
+1.  Note down it’s id to be passed to Stackdriver handler as destination for the
+    sink. It would be of the form
+    `bigquery.googleapis.com/projects/[PROJECT_ID]/datasets/[DATASET_ID]`
+1.  Give sink’s writer identity: cloud-logs@system.gserviceaccount.com BigQuery
+    data editor role in IAM
+
+#### GCS
+
+1.  Create a GCS bucket where you would like logs to get exported in BigQuery.
+1.  Note down it’s id to be passed to Stackdriver handler as destination for the
+    sink. It would be of the form `storage.googleapis.com/[BUCKET_ID`
+1.  Give sink’s writer identity: cloud-logs@system.gserviceaccount.com storage
+    object creator role in IAM
+
+#### Pub/Sub
+
+1.  Create a topic where you would like logs to get exported in Pub/Sub
+1.  Note down it’s id to be passed to Stackdriver handler as destination for the
+    sink. It would be of the form
+    `pubsub.googleapis.com/projects/[PROJECT_ID]/topics/[TOPIC_ID]`
+1.  Give sink’s writer identity: cloud-logs@system.gserviceaccount.com Pub/Sub
+    Publisher role in IAM
+
+### Setting up Stackdriver
+
+To configure exporting logs to Stackdriver, you have to write a handler for
+Stackdriver. Config proto for Stackdriver can be found
+[here](https://github.com/istio/istio/blob/master/mixer/adapter/stackdriver/config/config.proto).
+Handler is configured based on this proto.
+
+1.  Save following yaml file as `stackdriver.yaml`. Replace `<project_id>,
+    <sink_id>, <sink_destination>, <log_filter>` with their specific values.
+
+    ```yaml
+        apiVersion: "config.istio.io/v1alpha2"
+        kind: stackdriver
+        metadata:
+          name: handler
+          namespace: istio-system
+        spec:
+          # We'll use the default value from the adapter, once per minute, so we don't need to supply a value.
+          # pushInterval: 1m
+          # Must be supplied for the Stackdriver adapter to work
+          project_id: "<project_id>"
+          # One of the following must be set; the preferred method is `appCredentials`, which corresponds to
+          # Google Application Default Credentials. See:
+          #    https://developers.google.com/identity/protocols/application-default-credentials
+          # If none is provided we default to app credentials.
+          # appCredentials:
+          # apiKey:
+          # serviceAccountPath:
+          # Describes how to map Istio metrics into Stackdriver.
+          # Note: most of this config is copied over from prometheus.yaml to keep our metrics consistent across backends
+          metricInfo:
```

Comment:
do we want `metricInfo` here? I feel like that is beyond the scope of this Task, and makes it harder to reason about the logs work.

Reply:
Removed